File: javascript/loops.js
const iterations = process.argv[2];

function simulate() {
  const start = new Date().getTime()
  const sampleSize = 23

  let count = 0
  for (let i=0; i < iterations; i++) {
    const arr = new Array(365)
    for (let j=0; j < sampleSize; j++) {
      const rand = Math.floor(Math.random()*365)
      if (arr[rand]) {
        count++
        break
      } else {
        arr[rand] = 1
      }
    }
  }
  console.log(`iterations: ${iterations}`)
  console.log(`sample-size: ${sampleSize}`)
  const results = (count / iterations * 100).toFixed(2)
  console.log(`percent: ${results}`)
  const end = new Date().getTime()
  const diff = ((end-start) / 1000).toFixed(3)
  console.log(`seconds: ${diff}`)
}

simulate()

File: dart/loops.dart
import 'dart:math';

void main() {
  simulate();
}

void simulate() {
  final start = new DateTime.now().millisecondsSinceEpoch;
  const iterations = 1000 * 1000;
  const sampleSize = 23;

  var count = 0;
  final random = new Random();
  for (var i=0; i < iterations; i++) {
    final arr = [];
    for (var j=0; j < sampleSize; j++) {
      final number = random.nextInt(364);
      if (arr.contains(number)) {
        count++;
        break;
      } else {
        arr.add(number);
      }
    }
  }
  print('iterations: $iterations');
  print('sample-size: $sampleSize');
  final results = (count / iterations * 100).toStringAsFixed(2);
  print('percent: $results');
  final end = new DateTime.now().millisecondsSinceEpoch;
  final diff = ((end - start) / 1000).toStringAsFixed(3);
  print('seconds: $diff');
}
